chain/messagepool/check.go

```diff
@@ -32,14 +32,14 @@ func (mp *MessagePool) CheckMessages(ctx context.Context, protos []*api.MessageP
 // CheckPendingMessages performs a set of logical sets for all messages pending from a given actor
 func (mp *MessagePool) CheckPendingMessages(ctx context.Context, from address.Address) ([][]api.MessageCheckStatus, error) {
 	var msgs []*types.Message
-	mp.lk.Lock()
```

Comment:
~~`RLock` is more complex and can be much slower in some condition, I would not trust this without a benchmark.~~

Actually I see this is already an `sync.RWMutex`, I wouldn't trust this but this is not like this changing much compared to the current state so that probably fine.